pkg/staticnetworkconfig/generator.go

```diff
@@ -88,25 +91,61 @@ func (s *StaticNetworkConfigGenerator) createNMConnectionFiles(nmstateOutput str
 	return nil
 }
 
-// TODO - in case we will need additional formating, consider using https://github.com/go-ini/ini
 func (s *StaticNetworkConfigGenerator) formatNMConnection(nmConnection string) (string, error) {
-	start := strings.Index(nmConnection, "interface-name=")
-	if start == -1 {
-		msg := "Failed to find interface-name label in nmconnection string"
+	ini.PrettyFormat = false
+	cfg, err := ini.LoadSources(ini.LoadOptions{IgnoreInlineComment: true}, []byte(nmConnection))
+	if err != nil {
+		s.log.WithError(err).Errorf("Failed to load the ini format string %s", nmConnection)
+		return "", err
+	}
+	connectionSection := cfg.Section("connection")
+	intName := connectionSection.Key("interface-name")
+	if intName == nil {
+		msg := "interface-name key is not present in section connection"
 		s.log.Errorf("%s", msg)
 		return "", fmt.Errorf("%s", msg)
 	}
-	end := strings.Index(nmConnection[start:], "\n")
-	mac := nmConnection[start+len("interface-name=") : start+end]
+
+	mac := intName.String()
 	splitMac := []string{}
 	i := 0
 	for i < len(mac) {
 		splitMac = append(splitMac, mac[i:i+2])
 		i += 2
 	}
 	mac = strings.Join(splitMac, ":")
-	output := nmConnection[:start] + nmConnection[start+end+1:] + "\n[802-3-ethernet]\nmac-address=" + mac + "\n"
-	return output, nil
+
+	connectionSection.DeleteKey("interface-name")
+	_, err = connectionSection.NewKey("autoconnect", "true")
+	if err != nil {
+		s.log.WithError(err).Errorf("Failed to add autoconnect key to section connection")
+		return "", err
+	}
+	_, err = connectionSection.NewKey("autoconnect-priority", "1")
```

Comment:
I wasn't able to find an attribute named `autoconnect-priority` in https://developer.gnome.org/NetworkManager/stable/NetworkManager.conf.html
(search for first occurence of `autoconnect-`)

In addition, the same priority will be given to all of the interfaces of the same host.

Reply:
please, see here: https://developer.gnome.org/NetworkManager/stable/settings-connection.html
Regarding priority to different interfaces: we don't care about priority between interfaces, we care about priority between different connections for the same interface.  So, in case we have default connection that defines dhcp for all interfaces, we want our connections to have higher priority so that they will be picked up by NM